clusterloader2/pkg/measurement/common/network/performance-measurement-helper.go

```diff
@@ -0,0 +1,202 @@
+	if !strings.Contains(ratio, ratioSeparator) {
+		return false
+	}
+	return true
+}
+
+func validateClientServerPodNum(ratio string) (bool, error) {
+	var podNumber []string
+	podNumber = strings.Split(ratio, ratioSeparator)
+	clientPodNum, err := strconv.Atoi(podNumber[0])
+	if err != nil {
+		return false, err
+	}
+
+	serverPodNum, err := strconv.Atoi(podNumber[1])
+	if err != nil {
+		return false, err
+	}
+
+	if clientPodNum <= 0 || serverPodNum <= 0 {
+		return false, fmt.Errorf("invalid pod numbers")
+	}
+	return true, nil
+}
+
+func deriveClientServerPodNum(ratio string) (int, int, string) {
+	var podNumber []string
+	var clientPodNum, serverPodNum int
+
+	podNumber = strings.Split(ratio, ratioSeparator)
+	clientPodNum, _ = strconv.Atoi(podNumber[0])
+	serverPodNum, _ = strconv.Atoi(podNumber[1])
+	if clientPodNum == serverPodNum && clientPodNum == 1 {
+		return clientPodNum, serverPodNum, oneToOne
+	}
+	if (clientPodNum > serverPodNum) && serverPodNum == 1 {
+		return clientPodNum, serverPodNum, manyToOne
+	}
+	if clientPodNum == serverPodNum {
+		return clientPodNum, serverPodNum, manyToMany
+	}
+	return clientPodNum, serverPodNum, invalid
+}
+
+func (npm *networkPerformanceMeasurement) populateTemplate(podPair uniquePodPair, index int) map[string]interface{} {
+	return map[string]interface{}{
+		"index":                index,
+		"clientPodName":        podPair.sourcePodName,
+		"serverPodName":        podPair.destinationPodName,
+		"clientPodIP":          podPair.sourcePodIP,
+		"serverPodIP":          podPair.destinationPodIP,
+		"protocol":             npm.protocol,
+		"duration":             npm.testDuration,
+		"clientStartTimestamp": npm.startTimeStampForTestExecution,
+		"numberOfClients":      1}
+}
+
+func getUnusedPod(unusedPodList *[]workerPodData) (workerPodData, error) {
+	var unusedPod workerPodData
+	if len(*unusedPodList) == 0 {
+		return unusedPod, errors.New("unused pod list empty")
+	}
+	numOfPods := len(*unusedPodList)
+	// extract last pod of slice
+	unusedPod = (*unusedPodList)[numOfPods-1]
+	*unusedPodList = (*unusedPodList)[:numOfPods-1]
+	return unusedPod, nil
+}
+
+func getStartTimeStampForPod() int64 {
+	currTime := time.Now()
+	initDelay := time.Second * time.Duration(initialDelayForTestExecution)
+	futureTime := currTime.Add(initDelay).Unix()
+	return futureTime
+}
+
+func getUnit(metric string) string {
+	return metricUnitMap[metric]
+}
+
+func (npm *networkPerformanceMeasurement) closeCh() {
+	close(npm.podPairCh)
+}
+
+type float64Slice []float64
+
+func (p float64Slice) Len() int           { return len(p) }
+func (p float64Slice) Less(i, j int) bool { return p[i] < p[j] }
+func (p float64Slice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+func getPercentile(values float64Slice, percentile float64) float64 {
+	ps := []float64{percentile}
+
+	scores := make([]float64, len(ps))
+	size := len(values)
+	if size <= 0 {
+		return scores[0]
+	}
+
+	sort.Sort(values)
+	for i, p := range ps {
```

Comment:
? am I getting this wrong?
```
	ps := []float64{percentile}
```
ps always have one element, no?

Reply:
Initially this method was implemented keeping mind , we will pass array of percentile values (i.e 0.5,.75,.9 etc). Then i modified it a bit later due to some other reason. I have modified it now to take only 1 value.